### PR TITLE
Make slippi logo do a barrel roll on hover

### PR DESCRIPTION
**Pull request by rapito**
# It's quiet.. too quiet...
giphy

https://user-images.githubusercontent.com/1380104/129359530-2b834eda-3f0a-4a93-b352-5510408d7142.mov

### CODE REVIEW

**Comment by vinceau** [comment]
I love the idea but I don't love the implementation. Specifically this behaviour:

https://user-images.githubusercontent.com/8384734/129431044-9821caeb-16a6-41a1-8d01-662c0fb8b3ab.mov

**Comment by rapito** [comment]
> I love the idea but I don't love the implementation. Specifically this behaviour:
> 
>  Screen.Recording.2021-08-14.at.11.50.11.am.mov

I refactored it to take a different approach.

**Comment by vinceau** [comment]
> > I love the idea but I don't love the implementation. Specifically this behaviour:
> > Screen.Recording.2021-08-14.at.11.50.11.am.mov
> 
> I refactored it to take a different approach.

Does the new approach address the behaviour I mentioned before? If so, could you include in the PR a screengrab/video of the new behaviour when the mouse enters and leaves before the animation completes?

**Comment by rapito** [comment]
> > > I love the idea but I don't love the implementation. Specifically this behaviour:
> > > Screen.Recording.2021-08-14.at.11.50.11.am.mov
> > 
> > 
> > I refactored it to take a different approach.
> 
> Does the new approach address the behaviour I mentioned before? If so, could you include in the PR a screengrab/video of the new behaviour when the mouse enters and leaves before the animation completes?

Yes:


https://user-images.githubusercontent.com/1380104/132908692-ca12479b-a782-4314-8bd1-68b5e6f1400c.mov

It only rolls once now though. And that's fine :)

File: src/renderer/components/BouncingSlippiLogo.tsx [hunk 1]
```diff
@@ -23,13 +28,18 @@ const Outer = styled.div<{
 
 const Logo = styled.div<{
   size: string;
+  roll: boolean;
```
Comment (vinceau) [inline src/renderer/components/BouncingSlippiLogo.tsx:31]:
```suggestion
  animationState: "running" | "paused";
```
Maybe we can just pass the animation state in?

File: src/renderer/components/BouncingSlippiLogo.tsx [hunk 2]
```diff
@@ -23,13 +28,18 @@ const Outer = styled.div<{
...
 }>`
   background-image: url("${slippiLogo}");
   background-size: contain;
   background-repeat: no-repeat;
-  animation: ${bounceAnimation} 1.2s infinite;
+  animation: ${bounceAnimation} 1.2s infinite forwards,
+    ${barrelRollAnimation} 1s cubic-bezier(0.68, -0.55, 0.265, 1.55) alternate forwards;
+  animation-play-state: running, ${(props) => (props.roll ? "running" : "paused")};
```
Comment (vinceau) [inline src/renderer/components/BouncingSlippiLogo.tsx:38]:
```suggestion
  animation-play-state: running, ${(props) => props.animationState};
```

File: src/renderer/components/BouncingSlippiLogo.tsx [hunk 3]
```diff
@@ -23,13 +28,18 @@ const Outer = styled.div<{
...
   position: absolute;
-  height: ${(props) => props.size};
+  // logo is proportionally smaller to size by a factor of 0.75
+  // so we need to scale it down to avoid pivotal animations
```
Comment (vinceau) [inline src/renderer/components/BouncingSlippiLogo.tsx:41]:
```suggestion
```
I think I would prefer to keep comments outside of the `css` definition itself. If you want to comment maybe add it as javascript comments above the definition.

**Review by vinceau — CHANGES_REQUESTED** [review]

**Comment by rapito** [comment]
There.

**Review by vinceau — APPROVED** [review]
LGTM

**Pull request merged**